formulae/viewer/routes.py

```diff
@@ -0,0 +1,205 @@
+        if isinstance(collection, CtsWorkMetadata):
+            editions = [t for t in collection.children.values() if isinstance(t, CtsEditionMetadata)]
+            if len(editions) == 0:
+                raise UnknownCollection('{}'.format(collection.get_label()) + _l(' hat keine Edition.'))
+            objectId = str(editions[0].id)
+        formulae = current_app.picture_file[objectId]
+        passage_data = get_passage(objectId, '1')
+
+
+        #this viewer work when the library or archiv give an IIIF API for the external usage of theirs books
+        if "manifest" in formulae:
+            if "codex" in formulae:
+                codex=formulae["codex"]
+                manifest = url_for('viewer.static', filename=formulae["manifest"])
+                return current_app.config['nemo_app'].render(template='viewer::multiviewermirador.html', manifest=manifest
+                                                             ,objectId=objectId, codex=codex, passage_data=passage_data, url=dict())
+            else:
+                manifest = url_for('viewer.static', filename=formulae["manifest"])
+                with open((current_app.IIIFmapping+"/"+formulae["manifest"]), "r") as f:
+                    title = load(f)
+                codex=title["label"]
+                return current_app.config['nemo_app'].render(template='viewer::multiviewermirador.html', manifest=manifest
+                                                             ,objectId=objectId, codex=codex, passage_data=passage_data, url=dict())
+        else:
+            try:
+                view = int(view)
+            except:
+                #exception when in the url view contains another character than number
+                view=0
+                images = formulae["images"]
+                folios = formulae["folios"]
+                town = formulae['town']
+                codex = formulae['codex']
+                current_folios = folios[view]
+                flash( _('There is not an images for this formula. Showing the first page.'))
+                link_picture=current_app.IIIFserver +str(images[view])
+                return current_app.config['nemo_app'].render(template='viewer::multiviewer.html', picture=link_picture, objectId=objectId,
+                                                         current_view=view, total_views=len(images), passage_data=passage_data,
+                                                         url=dict(), town=town, codex=codex, current_folios=current_folios,folios=folios)
+            images = formulae['images']
+            folios = formulae['folios']
+            town = formulae['town']
+            codex = formulae['codex']
+
+            #avoid error when the number of view is over the maximum
+            if view > len(images)-1:
+
+                flash( _('There are not {} images for this formula. Showing the last page image instead.'.format(view)))
```

Comment:
Original messages should be in German. I would suggest:

`flash(_('Es gibt nur ') + str(len(images)) + _(' Bilder für diese Formel. Das letzte Bild wird hier gezeigt.'))`